File: clients/ios/Lextures/Core/LMS/BoardsGovernanceAdminLogic.swift
import Foundation

/// MOB.8 admin boards governance — menu gating + policy helpers.
enum BoardsGovernanceAdminLogic {
    static let rbacManagePermission = "global:app:rbac:manage"

    static func adminConsoleEnabled(_ features: MobilePlatformFeatures) -> Bool {
        features.ffMobileAdminConsole
    }

    static func canManage(permissions: [String]) -> Bool {
        permissions.contains(rbacManagePermission)
    }

    static func canView(features: MobilePlatformFeatures, permissions: [String]) -> Bool {
        adminConsoleEnabled(features)
            && features.ffMobileBoardsAdvanced
            && canManage(permissions: permissions)
    }

    static func shouldShowInMenu(features: MobilePlatformFeatures, permissions: [String]) -> Bool {
        canView(features: features, permissions: permissions)
    }

    static func webPath() -> String { "/admin/boards" }
}

File: clients/android/app/src/main/kotlin/com/lextures/android/core/lms/BoardsGovernanceAdminLogic.kt
package com.lextures.android.core.lms

import com.lextures.android.core.navigation.MobilePlatformFeatures

/** MOB.8 admin boards governance — menu gating + policy helpers. */
object BoardsGovernanceAdminLogic {
    const val RBAC_MANAGE_PERMISSION = "global:app:rbac:manage"

    fun adminConsoleEnabled(features: MobilePlatformFeatures): Boolean = features.ffMobileAdminConsole

    fun canManage(permissions: Collection<String>): Boolean =
        RBAC_MANAGE_PERMISSION in permissions

    fun canView(features: MobilePlatformFeatures, permissions: Collection<String>): Boolean =
        adminConsoleEnabled(features) &&
            features.ffMobileBoardsAdvanced &&
            canManage(permissions)

    fun shouldShowInMenu(features: MobilePlatformFeatures, permissions: Collection<String>): Boolean =
        canView(features, permissions)

    fun webPath(): String = "/admin/boards"
}
